posix: timer: timer kconfig option is not used to trigger compilation with cmake

**Describe the bug**
Currently `lib/posix/timer.c` is only built when `CONFIG_POSIX_CLOCK=y`, which is a bit unintuitive, since there is already a dedicated `CONFIG_TIMER` Kconfig option.

Please also mention any information which could help others to understand
the problem you're facing:
 - What target platform are you using? all
 - What have you tried to diagnose or workaround this issue? looked at the source
 - Is this a regression? No

**To Reproduce**
Steps to reproduce the behavior:
1. add `timer_create()` to `samples/hello_world`
2. west build -p auto -b qemu_riscv64 -t run samples/hello_world -- -DCONFIG_TIMER=y
3. See error

**Expected behavior**
I would expect that `timer.c` functions (`timer_create()`, etc) are visible to the application when `CONFIG_TIMER=y`

**Impact**
Annoyance

**Logs and console output**
```shell
: in function `main':
/Users/cfriedt/zephyrproject/zephyr/samples/hello_world/src/main.c:12: undefined reference to `timer_create'
```

**Environment (please complete the following information):**
 - OS: Linux
 - Toolchain: Zephyr SDK v0.16.4
 - Commit: cf14d4f1fd6d3049e3417d82ab18a2c9aa43b370

**Additional context**

--- a/lib/posix/CMakeLists.txt
+++ b/lib/posix/CMakeLists.txt
@@ -43,7 +43,6 @@ zephyr_library_sources_ifdef(CONFIG_POSIX_API perror.c)
 zephyr_library_sources_ifdef(CONFIG_POSIX_CLOCK clock.c)
 zephyr_library_sources_ifdef(CONFIG_POSIX_CLOCK nanosleep.c)
 zephyr_library_sources_ifdef(CONFIG_POSIX_CLOCK sleep.c)
-zephyr_library_sources_ifdef(CONFIG_POSIX_CLOCK timer.c)
 zephyr_library_sources_ifdef(CONFIG_POSIX_FS fs.c)
 zephyr_library_sources_ifdef(CONFIG_POSIX_MQUEUE mqueue.c)
 zephyr_library_sources_ifdef(CONFIG_POSIX_SIGNAL signal.c ${STRSIGNAL_TABLE_H})
@@ -58,6 +57,7 @@ zephyr_library_sources_ifdef(CONFIG_PTHREAD_RWLOCK rwlock.c)
 zephyr_library_sources_ifdef(CONFIG_POSIX_PRIORITY_SCHEDULING sched.c)
 zephyr_library_sources_ifdef(CONFIG_PTHREAD_IPC semaphore.c)
 zephyr_library_sources_ifdef(CONFIG_PTHREAD_SPINLOCK spinlock.c)
+zephyr_library_sources_ifdef(CONFIG_TIMER timer.c)
 
 zephyr_library_include_directories(
   ${ZEPHYR_BASE}/kernel/include

--- a/lib/posix/Kconfig.clock
+++ b/lib/posix/Kconfig.clock
@@ -3,9 +3,9 @@
 # SPDX-License-Identifier: Apache-2.0
 
 config POSIX_CLOCK
-	bool "POSIX clock, timer, and sleep APIs"
+	bool "POSIX clock and sleep APIs"
 	default y if POSIX_API
+	imply TIMER
 	depends on !NATIVE_LIBC
 	help
-	  This enables POSIX clock\_\*(), timer\_\*(), and \*sleep()
-	  functions.
+	  This enables POSIX clock\_\*() and \*sleep() functions.

